applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -1312,6 +1313,34 @@ void SmallStrainUPwDiffOrderElement::CalculateAll(MatrixType&        rLeftHandSi
     KRATOS_CATCH("")
 }
 
+std::vector<double> SmallStrainUPwDiffOrderElement::CalculateDerivativesOfSaturation(
+    const std::vector<double>& fluid_pressures, RetentionLaw::Parameters& RetentionParameters)
+{
+    std::vector<double> result;
+    std::transform(fluid_pressures.begin(), fluid_pressures.end(), mRetentionLawVector.begin(),
+                   std::back_inserter(result),
+                   [&RetentionParameters](double fluid_pressure, RetentionLaw::Pointer pRetentionLaw) {
+        RetentionParameters.SetFluidPressure(fluid_pressure);
+        return pRetentionLaw->CalculateDerivativeOfSaturation(RetentionParameters);
+    });
+
+    return result;
+}
+
+std::vector<double> SmallStrainUPwDiffOrderElement::CalculateDegreesOfSaturation(
+    const std::vector<double>& fluid_pressures, RetentionLaw::Parameters& RetentionParameters)
```

Comment:
The above comments also apply to this member function.

Reply:
Done